src/context-selection/category-option-combo-selector-bar-item/category-option-combo-selector-bar-item.js

```diff
@@ -0,0 +1,94 @@
+import i18n from '@dhis2/d2-i18n'
+import { SelectorBarItem } from '@dhis2/ui'
+import React, { useState } from 'react'
+import {
+    useCategoryOptionComboSelection,
+    useDataSetId,
+    useOrgUnitId,
+    usePeriodId,
+} from '../use-context-selection.js'
+import {
+    useDeselectOnDataSetChange,
+    useDeselectOnOrgUnitChange,
+    useDeselectOnPeriodChange,
+} from '../use-deselect/index.js'
+import CategoriesMenu from './categories-menu.js'
+import omit from './omit.js'
+import useCategoryCombination from './use-category-combination.js'
+import useSelected from './use-selected.js'
+import useValueLabel from './use-value-label.js'
+
+export default function CategoryOptionComboSelectorBarItem() {
+    const [open, setOpen] = useState(false)
+    const {
+        loadingCategoryCombination,
+        errorCategoryCombination,
+        categoryCombination,
+    } = useCategoryCombination()
+    const [dataSetId] = useDataSetId()
+    const [periodId] = usePeriodId()
+    const [orgUnitId] = useOrgUnitId()
+    const [, setCategoryOptionComboSelection] =
+        useCategoryOptionComboSelection()
+    const [selected, setSelected] = useSelected()
+    const valueLabel = useValueLabel()
+    const deselectAll = () => {
+        setSelected({})
+        setCategoryOptionComboSelection([])
+    }
+
+    const onChange = ({ selected: value, categoryId }) => {
+        if (!value) {
+            const nextSelected = omit(categoryId, selected)
+            setSelected(nextSelected)
+            return
+        }
+
+        const nextSelected = { ...selected, [categoryId]: value }
+        const nextCategoryOptionComboSelection = Object.entries(
+            nextSelected
+        ).map(([key, value]) => `${key}:${value}`)
+
+        setSelected(nextSelected)
+        setCategoryOptionComboSelection(nextCategoryOptionComboSelection)
+    }
+
+    useDeselectOnDataSetChange(deselectAll)
+    useDeselectOnOrgUnitChange(deselectAll)
+    useDeselectOnPeriodChange(deselectAll)
```

Comment:
Could this not be replaced with something like:

```
useEffect(() => {
  if (selectedNotCleared || categoryOptionComboSelectionNotCleared) {
    setSelected({})
    setCategoryOptionComboSelection([])
  }
}, [dataSetId, periodId, orgUnitId])
```

Current abstraction makes things harder to follow in my opinion.

Reply:
We talked about this. I changed it a bit, but not as proposed. Feel free to propose another suggestion, I'll mark all other comments talking about this as resolved